packages/kbn-esql-validation-autocomplete/src/autocomplete/helper.ts

```diff
@@ -55,3 +55,11 @@ export function getParamAtPosition(
 ) {
   return params.length > position ? params[position] : minParams ? params[params.length - 1] : null;
 }
+
+export function getQueryForFields(queryString: string, commands: ESQLCommand[]) {
+  // If there is only one source command and it does not require fields, do not
+  // fetch fields, hence return an empty string.
+  return commands.length === 1 && ['from', 'row', 'show'].includes(commands[0].name)
```

Comment:
I can change, but the reasoning is: to be conservative.

This way it works correctly going forward. If a new command is added, it might load the fields unnecessarily but if that command does indeed need the fields, the fields will be loaded.

Reply:
If the new command is not added manually by us on the validation and autocomplete system, loading the fields will be useless though. What I want to say is that allowing it for every new command doesnt give any advantages rather than loading the fields at every keystroke.